move/sources/food.move

```diff
@@ -0,0 +1,161 @@
+        let description = string::utf8(FOOD_COLLECTION_DESCRIPTION);
+        let name = string::utf8(FOOD_COLLECTION_NAME);
+        let uri = string::utf8(FOOD_COLLECTION_URI);
+
+        collection::create_unlimited_collection(
+            creator,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+    }
+
+    /// Creates the food token as fungible token.
+    fun create_food_token(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        fungible_asset_name: String,
+        fungible_asset_symbol: String,
+        icon_uri: String,
+        project_uri: String,
+    ) {
+        let collection = string::utf8(FOOD_COLLECTION_NAME);
+        let constructor_ref = token::create_named_token(
+            creator,
+            collection,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+
+        let token_signer = object::generate_signer(&constructor_ref);
+
+        // Creates the fungible asset.
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            &constructor_ref,
+            option::none(),
+            fungible_asset_name,
+            fungible_asset_symbol,
+            0,
+            icon_uri,
+            project_uri,
+        );
+        let fungible_asset_mint_ref = fungible_asset::generate_mint_ref(&constructor_ref);
+        let fungible_asset_burn_ref = fungible_asset::generate_burn_ref(&constructor_ref);
+
+        // Publishes the FoodToken resource with the refs.
+        let food_token = FoodToken {
+            fungible_asset_mint_ref,
+            fungible_asset_burn_ref,
+        };
+        move_to(&token_signer, food_token);
+    }
+
+    public fun get_food_token_address(): address {
+        let food_token_address = token::create_token_address(
+            &@aptogotchi,
+            &string::utf8(FOOD_COLLECTION_NAME),
+            &string::utf8(FOOD_NAME),
+        );
+
+        food_token_address
+    }
+
+    public fun mint_food(user: &signer, amount: u64) acquires FoodToken {
+        let food_token= borrow_global<FoodToken>(get_food_token_address());
+        let fungible_asset_mint_ref = &food_token.fungible_asset_mint_ref;
+        let fa = fungible_asset::mint(fungible_asset_mint_ref, amount);
+        primary_fungible_store::deposit(address_of(user), fa);
+    }
+
+    public fun burn_food(user: &signer, amount: u64) acquires FoodToken {
+        let food = object::address_to_object<FoodToken>(get_food_token_address());
+        let metadata = object::convert<FoodToken, Metadata>(food);
+        let food_token= borrow_global<FoodToken>(get_food_token_address());
+        let from_store = primary_fungible_store::ensure_primary_store_exists(signer::address_of(user), metadata);
+        fungible_asset::burn_from(&food_token.fungible_asset_burn_ref, from_store, amount);
```

Comment:
You can use https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/primary_fungible_store.move#L162 instead